### DISCUSSION TITLE
Lesson 2 - ParserError: Identifier-start is not allowed at end of a number

### DISCUSSION THREAD

**Discussion by hissho** [Cyfrin/foundry-full-course-cu#1226] [Q&A] [answered]
Hi all,
I just did a simple test with the "address" variable and to my surprise, it didn't work as expected:

address myWallet1 = 0xekf4378550hh7jit483957983fdkslfjdkrewr19fy;
address myWallet2 = 0xec61b9e42a4693F4C44f616c31DB9C2D1790cE9C;

I got an error message after hitting the "compile" button in Remix:
**ParserError: Identifier-start is not allowed at end of a number.**

"0xekf4378550hh7jit483957983fdkslfjdkrewr19fy" was what I randomly punched in and "0xec61b9e42a4693F4C44f616c31DB9C2D1790cE9C" is a MetaMask address.

So why did I get this error message and how can I correct it?

Thanks!

**Comment by ghost** [comment, marked as answer]
The problem is that `myWallet1` it's not a valid Ethereum address. 

Addresses in Ethereum are a 20-bytes hexadecimal string, this means that the allowed characters range from `0` to `9` and `a` to `f` (case insensitive). 

If you're just testing transferring ether from one account to another, just create a new address in Metamask. If you don't know how to add new accounts just follow the steps in the following link:

https://support.metamask.io/hc/en-us/articles/360015289452-How-to-add-accounts-in-your-wallet#:~:text=In%20wallet%20view%2C%20tap%20the,later%2C%20use%20the%20instructions%20here.

**Comment by hissho** [comment]
Thanks pacelliv. 

What you said makes sense although after taking another look, I'm baffled again now :)

1) > the allowed characters range from 0 to 9 and a to f (case insensitive).

if that's the case, why "x" is allowed?

2) > Addresses in Ethereum are a 20-bytes hexadecimal string

I counted and there're 42 bytes in address2, which is a valid address...so what did you mean exactly by "20-bytes"?

3) I manually modified address1 and now it has become: 
**0xec4378b550dd7e483957983fdfde19f1D5ac9d21**

I got another error message when compiling the code:
> SyntaxError: This looks like an address but has an invalid checksum. Correct checksummed address: "0xEC4378b550Dd7E483957983fDfDe19f1D5aC9D21". If this is not used as an address, please prepend '00'. 

Where can I get more info about valid/invalid checksum?

and what's the difference between "0x" and "00"?

Thanks and happy holidays!

**Reply by ghost** [reply]
1. > if that's the case, why "x" is allowed?

Correct, no letter after **f** is allowed.

2. > I counted and there're 42 bytes in address2, which is a valid address...so what did you mean exactly by "20-bytes"?

A byte is represented with 2 hexadecimals characters, each pair without considering the `0x` prefix is a byte. The prefix is only used to indicate the data is a hex string.

3. About Checksum:

High level explanation: https://docs.alchemy.com/docs/how-to-handle-checksum-addresses

Deeper explanation: https://medium.com/coinmonks/ethereum-checksummed-addresses-importance-and-implementation-eef74aa3ae18

Happy Holidays :)

**Reply by hissho** [reply]
Thanks bro, for answering the questions on a holiday!

All the best...
